.github/workflows/black.yaml

```diff
@@ -0,0 +1,17 @@
+name: lint black
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: psf/black@stable
+        with:
+          options: "--check --verbose --diff --line-length 110 --target-version py38"
```

Comment:
@TallJimbo I've turned on black in this repo to see how we get on. These options seemed reasonable.

Reply:
and I realized I could put those options in the `pyproject.toml` file.

Reply:
We probably want `isort` in the same action; here's what I did recently for the combo:

https://github.com/lsst-dm/cell_coadds/blob/main/.github/workflows/formatting.yaml
https://github.com/lsst-dm/cell_coadds/blob/main/pyproject.toml

Reply:
Okay done. I've also committed a pre-commit config file.